harness/determined/pytorch/lightning/_adapter.py

```diff
@@ -1,470 +1,469 @@
-            self._pls.lm.on_before_zero_grad(opt)
-            opt.zero_grad()
-
-            opt_metrics.append(metrics)
-            with monkey_patch(
-                self._pls.lm, "optimizers", lambda *args, **kwargs: self._pls.optimizers
-            ):
-                self._pls.lm.untoggle_optimizer(opt_idx)
-
-        self._pls.lm.on_train_batch_start(batch, batch_idx)
-
-        # report metrics accounting for duplicate metric names
-        # across multiple optimizers
-        metric_names: List[str] = []
-        for opt_metric_dict in opt_metrics:
-            metric_names += opt_metric_dict.keys()
-        duplicate_metrics = filter_duplicates(metric_names)
-
-        agg_metrics = {}
-        for opt_idx, opt_metric_dict in enumerate(opt_metrics):
-            for m_name, m_value in opt_metric_dict.items():
-                if m_name in duplicate_metrics:
-                    m_name = f"opt{opt_idx}_{m_name}"
-                agg_metrics[m_name] = m_value
-        return agg_metrics
-
-    def evaluate_batch(self, batch: TorchData, batch_idx: int) -> Dict[str, Any]:
-        """
-        evaluate_batch implements the evaluate_batch interface from PyTorchTrial using user provided
-        lightning_module.
-
-        """
-        type(self._pls.lm).global_step = batch_idx  # type: ignore
-        self._pls.lm.on_validation_batch_start(batch, batch_idx, dataloader_idx=0)
-        rv = self._pls.lm.validation_step(batch, batch_idx=batch_idx)
-        self._pls.lm.on_validation_batch_end(rv, batch, batch_idx, dataloader_idx=0)
-
-        metrics = None
-        if rv is None:
-            metrics = {}
-        elif not isinstance(rv, dict):
-            metrics = {"loss": rv}
-        else:
-            metrics = rv
-        return metrics
-
-    @abstractmethod
-    def build_training_data_loader(self) -> DataLoader:
-        """
-        Defines the data loader to use during training.
-
-        Must return an instance of :py:class:`determined.pytorch.DataLoader`.
-
-        If you're using ``LightningDataModule`` this could be as simple as:
-
-        .. code-block:: python
-
-            self.dm.setup()
-            dl = self.dm.train_dataloader()
-            return DataLoader(dl.dataset, batch_size=dl.batch_size,
-                             num_workers=dl.num_workers)
-
-
-        """
-        pass
-
-    @abstractmethod
-    def build_validation_data_loader(self) -> DataLoader:
-        """
-        Defines the data loader to use during validation.
-
-        Must return an instance of :py:class:`determined.pytorch.DataLoader`.
-
-        If you're using ``LightningDataModule`` this could be as simple as:
-
-        .. code-block:: python
-
-            self.dm.setup()
-            dl = self.dm.val_dataloader()
-            return DataLoader(dl.dataset, batch_size=dl.batch_size,
-                             num_workers=dl.num_workers)
-
-        """
-        pass
+import inspect

```

Comment:
Looks like this entire file was converted to windows line endings, can you fix that?